emote/models/model.py

```diff
@@ -175,43 +187,143 @@ def load(self, load_dir: str) -> None:
         self.model.load(load_dir)
 
 
-class ModelLoss(LossCallback):
-    """Trains a dynamic model by minimizing the model loss
-
-    Arguments:
-        dynamic_model (DynamicModel): A dynamic model
-        opt (torch.optim.Optimizer): An optimizer.
-        lr_schedule (lr_scheduler, optional): A learning rate scheduler
-        max_grad_norm (float): Clip the norm of the gradient during backprop using this value.
-        name (str): The name of the module. Used e.g. while logging.
-        data_group (str): The name of the data group from which this Loss takes its data.
-    """
-
+class DeterministicModel(nn.Module):
     def __init__(
         self,
-        *,
-        model: DynamicModel,
-        opt: optim.Optimizer,
-        lr_schedule: Optional[optim.lr_scheduler._LRScheduler] = None,
-        max_grad_norm: float = 10.0,
-        name: str = "dynamic_model",
-        data_group: str = "default",
+        in_size: int,
+        out_size: int,
+        device: torch.device,
+        hidden_size: int = 256,
+        num_hidden_layers: int = 4,
     ):
-        super().__init__(
-            name=name,
-            optimizer=opt,
-            lr_schedule=lr_schedule,
-            network=model,
-            max_grad_norm=max_grad_norm,
-            data_group=data_group,
-        )
-        self.model = model
+        super().__init__()
+        self.in_size = in_size
+        self.out_size = out_size
+        self.device = torch.device(device)
+
+        network = [
+            nn.Sequential(
+                nn.Linear(in_size, hidden_size),
+                nn.BatchNorm1d(hidden_size),
+                nn.ReLU(),
+            )
+        ]
+        for _ in range(num_hidden_layers - 1):
+            network.append(
+                nn.Sequential(
+                    nn.Linear(hidden_size, hidden_size),
+                    nn.BatchNorm1d(hidden_size),
+                    nn.ReLU(),
+                )
+            )
+        network.append(nn.Sequential(nn.Linear(hidden_size, out_size)))
+        self.network = nn.Sequential(*network).to(self.device)
+        self.init_param()
+
+    def init_param(self):
```

Comment:
This seems like an anti-pattern, why do you need a separate function? It's literally below.

Reply:
Fair point!
However, it should be also possible to use the init_param function by other NN models. So, I moved the function to utils.model. I also moved truncated_normal_init to utils.model for the same reason.